### DISCUSSION TITLE
How to setup CORS in GOA?

### DISCUSSION THREAD

**Discussion by arif2009** [goadesign/goa#3007] [General]
Where and how to allow different origins in GOA?

**Comment by raphael** [comment]
This is done using the CORS [plugin](https://github.com/goadesign/plugins), see the [README](https://github.com/goadesign/plugins/tree/v3/cors) for details.

**Reply by arif2009** [reply]
@raphael Thanks for your help 🙏 

In your README documentation link, I think you need to change:

```
cors.Origin("*.domain.com", func() {
    Headers("X-Shared-Secret", "X-Api-Version")
    MaxAge(100)
    Credentials()
})
```
to
```
cors.Origin("*.domain.com", func() {
    cors.Headers("X-Shared-Secret", "X-Api-Version")
    cors.MaxAge(100)
    cors.Credentials()
})
```

**Reply by raphael** [reply]
Makes sense, care to submit a PR?

**Reply by arif2009** [reply]
Thanks. I can do that.

**Reply by gregwebs** [reply]
@arif2009 what error are you getting? The source is here: https://github.com/goadesign/plugins/blob/v3/cors/dsl/cors.go

**Reply by arif2009** [reply]
@gregwebs I am not getting any errors. I just wanted to know "how to allow different origins". And I got my answer and it was solved.
